### ISSUE TITLE
[feat] Handle or warn when the user's public base path is not empty

### ISSUE THREAD

**Issue by sansseriff** [tauri-apps/tauri#10872]
### Describe the problem

When using vite, the user may define a [public base path](https://v2.vitejs.dev/guide/build#public-base-path) that is prepended the the url where the app is served in `dev` or in `preview`. 

When a custom base path is specified in `vite.config.ts`, `npm run tauri dev` works as expected, but `npm run tauri build` just shows a white screen. Upon enabling developer tools on the build, you see that the webview is complaining about MIME types of css and javascript files. So though this looks at first like a MIME type issue, I think it's just that the webview is looking for the js and css at the wrong path. 

### Describe the solution you'd like

Its possible this is all expected behavior. But for me as a newcomer, it led to a bug that took a long time to diagnose. 

Perhaps tauri could be updated to handle non-empty public base paths. Or warning messages could be added somewhere for diagnosing this. It's possible [other people have had the same issue](https://github.com/tauri-apps/tauri/issues/4098#issuecomment-1194715084). 

For anyone else searching base on error messages:
In the webview devtools console I was seeing: `TypeError: 'text/html' is not a valid JavaScript MIME type.` and `Did not parse stylesheet at 'tauri://localhost/<custom_base_path>/assets/index-D5V9wA_-.css' because non CSS MIME types are not allowed in strict mode.`

And my solution was to comment out the `base:` option in `defineConfig()` in `vite.config.ts`

**Comment by amrbashir** [comment]
This is more of a configuration error or mismatch, we expect `distDir` (or `frontendDist` in v2), to be pointing to a directory that contains the final HTML,CSS,JS script files, that should looks something like this:
```
dist/
  |__ index.html
  |__ script.js
  |__ style.css
```

where `index.html` references `script.js` and `style.css` relative to its path, e.g `/script.js` and `/style.css`

However if you configure your frontend build tool to specify a different base for your application, for instance the `base` option in vite, it will make `index.html` references `script.js` and `style.css` using a different path than expected, e.g `/base/script.js` and `/base/style.css` and so the `dist` directory above is invalid, and instead it should look something like this:
```
dist/
  |__ index.html
  |__ base/
    |__ script.js
    |__ style.css
``` 


tldr; There is a numerous of frontend build tools and tauri can't cover all of their configuration options, if you change your frontend config, you'll need to modify tauri.conf.json to account for your changes.
